### ISSUE TITLE
Refactor the authentication module `Connection.psm1` for scalability

### ISSUE THREAD

**Issue by buidav** [cisagov/ScubaGear#1233]
# 💡 Summary #

The old authentication works and is reliable but could use a refactor to better scale for additional authentication parameters, cross product authentication, and possibly for PowerShell 7 module switching purposes. 

Will need to break out each individual authentication call into their own custom function and since now to config file defaults we won't be duplicating as much code if this was done earlier. 

## Motivation and context ##

Clean up the old code in `Connection.psm1` for scaling for additional features. 
#1208 
#88 

## Implementation notes ##

Refactor Connection.psm1 to break out each authentication call into their own separate function, with room for scaling for additional authentication methods and cross product authentication when necessary. 

> Note that Exchange Online and Defender authentication functions were broken out into their own module because of an early technical reason with a Defender policy. 

## Acceptance criteria ##
Authentication code is refactored for better scaling for new features.

**Comment by FollyBeachGurl** [comment]
Can't work on this until #1208 is unblocked.  This issue is on HOLD.

**Comment by tkol2022** [comment]
Chatted with @buidav offline about this. We can close it out since the module has been somewhat refactored recently to better organize auth functionality into helpers.
